src/vs/workbench/contrib/notebook/browser/notebookServiceImpl.ts

```diff
@@ -333,7 +334,7 @@ export class NotebookOutputRendererInfoStore {
 				const rendererExtId = renderer.extensionId.value;
 				const reuseScore = preferred === renderer.id
 					? ReuseOrder.PreviouslySelected
-					: rendererExtId === notebookExtId || RENDERER_EQUIVALENT_EXTENSIONS.get(rendererExtId)?.has(notebookExtId!)
+					: rendererExtId === notebookExtId || RENDERER_EQUIVALENT_EXTENSIONS.get(rendererExtId)?.has(notebookId!)
```

Comment:
The interactive window doesn't have an extension id associated with it, so I used the ```notebookProviderInfo.id``` instead.